### PR TITLE
Implement support for SARA-R4 module.

### LINKED ISSUE #1033: SARA-R4 support
Implement SARA-R4 support.

### PR DESCRIPTION
**Pull request by brentpicasso**
Implement support for SARA-R4 module, and optimize re-connect logic. 

Resolves #1033  #1032.

### CODE REVIEW

File: src/devices/cellular.c [hunk 1]
```diff
@@ -419,9 +441,11 @@ int cellular_init_connection(DeviceConfig *config, millis_t * connected_at)
         /* This is sane since DeviceConfig is typedef'd as a serial_buffer */
         struct serial_buffer *sb = (struct serial_buffer*) config;
 
-        const int init_status = cellular_init_modem(sb);
-        if (DEVICE_INIT_SUCCESS != init_status)
-                return DEVICE_INIT_FAIL;
+        if (hard_init) {
+                const int init_status = cellular_init_modem(sb);
```
Comment (stieg) [inline src/devices/cellular.c]: So we only initialize now if hard init is set?  We otherwise don't init?  Seems off.

File: src/gsm/gsm.c [hunk 2]
```diff
@@ -38,7 +38,7 @@ bool gsm_ping_modem(struct serial_buffer *sb)
 
         serial_buffer_reset(sb);
         serial_buffer_append(sb, "AT");
-        const size_t count = cellular_exec_cmd(sb, READ_TIMEOUT, msgs,
+        const size_t count = cellular_exec_cmd(sb, READ_TIMEOUT * 2, msgs,
```
Comment (stieg) [inline src/gsm/gsm.c]: Why not just increase READ_TIMEOUT and not have unexpected behavior?

File: src/devices/cellular.c (hunk 1 above)
Comment (brentpicasso) [reply src/devices/cellular.c]: We pass in a flag for hard initialization - this is controlled by the outside routine (connectivity task). 

If the hard_init is false, we bypass the power cycle routine, which will speed up cellular acquisition. 

The gist is, if we lose connection, we give the system a chance to quickly re-connect. 

The logic in the connectivity task will do a 'fast init' (i.e. hard_init = false) up to X amount of times before it falls back to power cycling.

File: src/gsm/gsm.c (hunk 2 above)
Comment (brentpicasso) [reply src/gsm/gsm.c]: I think that's  good idea. fixed.

**Pull request merged**